app/models/sales_log.rb

```diff
@@ -348,12 +364,48 @@ def beds_for_la_sale_range
 
   def ownership_scheme
     case ownershipsch
-    when 1
-      "shared ownership"
-    when 2
-      "discounted ownership"
-    when 3
-      "outright sale"
+    when 1 then "shared ownership"
+    when 2 then "discounted ownership"
+    when 3 then "outright sale"
+    end
+  end
+
+private
+
+  def reset_invalidated_dependent_fields!
+    super
+
+    reset_derived_questions!
+  end
+
+  def reset_derived_questions!
```

Comment:
any reason this is a bang method compared to lettings_log's not being one?

Reply:
I believe it should be a bang method because it is !dangerous! ie it makes changes to the model.
I would suggest that the lettings log method also be updated in this way.

Reply:
go on then